### PR TITLE
[C-2987] Add UserGeneratedText

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Adds UserGeneratedText using Linkify, and ensuring text is consistently formatted.
Improves a bunch of other components to make this work.

TODO: There is one more linkify instance in messages @rickyrombo leaving that for you to make sure it's done properly. happy to try my hand though

### How Has This Been Tested?

Checked every usage multiple times

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-2987-user-generated-text

File: packages/web/src/components/user-generated-text/UserGeneratedText.module.css [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.root {
+  white-space: pre-line;
+  /* user-select: none; */
```
Comment (rickyrombo) [inline packages/web/src/components/user-generated-text/UserGeneratedText.module.css:3]: nit: comment

File: packages/web/src/components/user-generated-text/UserGeneratedText.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+  const children =
+    typeof childrenProp === 'string'
+      ? squashNewLines(childrenProp)
+      : childrenProp
+
+  return (
+    <Linkify
+      options={options}
+      // @ts-expect-error
```
Comment (rickyrombo) [inline packages/web/src/components/user-generated-text/UserGeneratedText.tsx:80]: why is this needed?

**Review by rickyrombo — APPROVED** [review]
This is awesome, killer work! Love that this has more deletions than additions <3

I can make the DMs use this in a follow up, do you mind making a ticket?

File: packages/web/src/components/user-generated-text/UserGeneratedText.module.css (hunk 1 above)
Comment (dylanjeffers) [reply packages/web/src/components/user-generated-text/UserGeneratedText.module.css:3]: nice catch

File: packages/web/src/components/user-generated-text/UserGeneratedText.tsx (hunk 2 above)
Comment (dylanjeffers) [reply packages/web/src/components/user-generated-text/UserGeneratedText.tsx:80]: linkify's "as" isn't as generic as it should be basically :/

Comment (rickyrombo) [reply packages/web/src/components/user-generated-text/UserGeneratedText.tsx:80]: sad :( they should have used asChild :joy:

Comment (dylanjeffers) [reply packages/web/src/components/user-generated-text/UserGeneratedText.tsx:80]: after doing this pr, i have seen the light with asChild though, i think for harmony we may want to go with that instead
